storybookjs/storybook
# Refactor framework-independent core library

Issue: #2240 

## What I did

**Work in progress**: first proof of concept for discussion

- create `@storybook/core` library
- move most common code into library
- add some per-framework configuration for decorators

Still to-do:
- [x] Update RN
- [ ] Refactor RN more aggressively
- [ ] Refactor manager code
- [ ] Refactor server code
- [ ] Update tests/docs

## How to test

```
yarn bootstrap --core
```

```
cd examples/cra-kitchen-sink
yarn storybook
cd ../vue-kitchen-sink
yarn storybook
cd ../angular-cli
yarn storybook
cd ../..
yarn test
```

Is this testable with jest or storyshots?

Does this need a new example in the kitchen sink apps?

Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.
